src/runtime/propensity-server.js

```diff
@@ -80,6 +81,29 @@ export class PropensityServer {
     return this.clientId_;
   }
 
+  /**
+   * @return {string}
+   * @private
+   */
+  getReferrer_() {
+    return this.win_.document.referrer;
+  }
+
+  /**
+   * @private
+   * @param {string} url
+   * @return {string}
+   */
+  propensityUrl_(url) {
+    url = url + '&u_tz=240&v=' + this.version_;
+    const clientId = this.getClientId_();
+    if (clientId) {
+      url = url + '&cookie=' + clientId;
+    }
+    url = url + '&cdm=' + parseUrl(this.getReferrer_()).origin;
```

Comment:
there should never be an empty string, since it's called internally from another method. I do believe in doing null checks for APIs facing externally. Also, since this is an internal helper function, we don't test it directly. in tests we only verify APIs like sendEvent, sendSubscriptionState and getPropensity. Testing this method would amount to test implementation, not behavior.

Reply:
Ah I see your question was for parseUrl?
The return is non nulllable. But origin field can be null. Adding tests